src/nmea.cpp

```diff
@@ -0,0 +1,1043 @@
+
+		if (bufptr && *(++bufptr) != ',') { alt = strtof(bufptr, &endp); bufptr = endp; }
+
+		if (ns == 'S') {
+			lat = -lat;
+		}
+
+		if (ew == 'W') {
+			lon = -lon;
+		}
+
+		/* convert from degrees, minutes and seconds to degrees */
+		_gps_position->lat = static_cast<int>((int(lat * 0.01) + (lat * 0.01 - int(lat * 0.01)) * 100.0 / 60.0) * 10000000);
+		_gps_position->lon = static_cast<int>((int(lon * 0.01) + (lon * 0.01 - int(lon * 0.01)) * 100.0 / 60.0) * 10000000);
+		_gps_position->hdop = hdop;
+		_gps_position->alt = static_cast<int>(alt * 1000);
+		sat_num_gns = static_cast<int>(num_of_sv);
+
+		if (!_POS_received && (_last_POS_timeUTC < utc_time)) {
+			_last_POS_timeUTC = utc_time;
+			_POS_received = true;
+		}
+		_ALT_received = true;
+		_SVNUM_received = true;
+
+		// mavlink_log_info(&mavlink_log_pub, "GNS> posMode %s/sv_num %d/ hdop %.2f",
+		// (char*)(pos_Mode),
+		// (int)(sat_num_gns),
+		// (double)(hdop));
+
+	}	else if ((memcmp(_rx_buffer + 3, "RMC,", 4) == 0) && (uiCalcComma >= 11)) {
+
+		/*
+		Position, velocity, and time
+		The RMC string is:
+
+		$xxRMC,time,status,lat,NS,long,EW,spd,cog,date,mv,mvEW,posMode,navStatus*cs<CR><LF>
+		The Talker ID ($--) will vary depending on the satellite system used for the position solution:
+		$GNRMC,092721.00,A,2926.688113,N,11127.771644,E,0.780,,200520,,,D,V*1D
+
+		GPRMC message fields
+		Field	Meaning
+		0	Message ID $GPRMC
+		1	UTC of position fix
+		2	Status A=active or V=void
+		3	Latitude
+		4	Longitude
+		5	Speed over the ground in knots
+		6	Track angle in degrees (True)
+		7	Date
+		8	Magnetic variation in degrees
+		9	The checksum data, always begins with *
+		*/
+		double utc_time = 0.0;
+		char Status = 'V';
+		double lat = 0.0, lon = 0.0;
+		float ground_speed_K = 0.f;
+		float track_true = 0.f;
+		int nmea_date = 0;
+		float Mag_var = 0.f;
+		char ns = '?', ew = '?';
+		NMEA_UNUSED(Mag_var);
+
+		_decode_flags |= (int)NMEADecodeFlags::got_rmc;
+
+		if (bufptr && *(++bufptr) != ',') { utc_time = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { Status = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { lat = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { ns = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { lon = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { ew = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { ground_speed_K = strtof(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { track_true = strtof(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { nmea_date = static_cast<int>(strtol(bufptr, &endp, 10)); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { Mag_var = strtof(bufptr, &endp); bufptr = endp; }
+
+		if (ns == 'S') {
+			lat = -lat;
+		}
+
+		if (ew == 'W') {
+			lon = -lon;
+		}
+
+		if (Status == 'V') {
+			return 0;
```

Comment:
I don't think it is right to return from RMC there is another information in it below, I suggest to just clear fix_type without return:
`_gps_position->fix_type = 0;`

Reply:
Good idea and follows the same approach as in other quality indicators.